sympy/physics/continuum_mechanics/beam.py

```diff
@@ -0,0 +1,436 @@
+        >>> b.boundary_conditions['moment']
+        [(0, 4), (4, 0), (4, 3), (5, 0)]
+
+        """
+        for bcs in m_bcs:
+            self._boundary_conditions['moment'].append(bcs)
+
+    def apply_slope_boundary_conditions(self, *s_bcs):
+        """
+        Takes only the slope boundary conditions as input.
+
+        Examples
+        ========
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> b.apply_boundary_conditions(moment=[(0, 4), (4, 0)], deflection=[(0, 2)], slope=[(0, 1)])
+        >>> b.boundary_conditions
+        {'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
+        >>> b.apply_slope_boundary_conditions((4, 3), (5, 0))
+        >>> b.boundary_conditions['slope']
+        [(0, 1), (4, 3), (5, 0)]
+
+        """
+        for bcs in s_bcs:
+            self._boundary_conditions['slope'].append(bcs)
+
+    def apply_deflection_boundary_conditions(self, *d_bcs):
+        """
+        Takes only the slope boundary conditions as input.
+
+        Examples
+        ========
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> b.apply_boundary_conditions(moment=[(0, 4), (4, 0)], deflection=[(0, 2)], slope=[(0, 1)])
+        >>> b.boundary_conditions
+        {'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
+        >>> b.apply_deflection_boundary_conditions((4, 3), (5, 0))
+        >>> b.boundary_conditions['deflection']
+        [(0, 2), (4, 3), (5, 0)]
+
+        """
+        for bcs in d_bcs:
+            self._boundary_conditions['deflection'].append(bcs)
+
+    def apply_load(self, value, start, order):
+        x = self.variable
+        value = sympify(value)
+        start = sympify(start)
+        order = sympify(order)
+
+        self._load += value*SingularityFunction(x, start, order)
+
+    def load_distribution(self):
+        """
+        Returns a Singularity Function expression which represents
+        the load distribution curve of the Beam object.
+
+        Examples
+        ========
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> b.apply_load(order=-2, value=-3, start=0)
+        >>> b.apply_load(order=-1, value=4, start=2)
+        >>> b.apply_load(order=2, value=-2, start=3)
+        >>> b.apply_boundary_conditions(moment=[(0, 4), (4, 0)], deflection=[(0, 2)], slope=[(0, 1)])
+        >>> b.load_distribution()
+        -3*SingularityFunction(x, 0, -2) + 4*SingularityFunction(x, 2, -1) - 2*SingularityFunction(x, 3, 2)
+        """
+        return self._load
```

Comment:
I think the idea of doing this will not be that good. 
```py
b.load += -3*SingularityFunction(x, 0, -1)
```
Because 

-  We can't test whether the input is a `SingularityFunction` object. The one which is passed here is `Mul` object.
- We have one setter function know as `apply_load`. 

I think making load as a property would be the only good idea. Let's leave the idea of making it as setter too.